src/timeout.ts

```diff
@@ -0,0 +1,129 @@
+import { clearTimeout, setTimeout } from 'timers';
+
+import { MongoError, MongoInvalidArgumentError } from './error';
+import { noop } from './utils';
+
+/** @internal */
+export class CSOTError extends MongoError {
```

Comment:
```suggestion
export class MongoDBTimeoutError extends MongoError {
```

We're in this tight spot with naming 😅 every timeout we fire is "client side" but since CSOT refers to the `timeoutMS` feature we can change this name and this error is going to serve as a communication of what kind of timeout we fired.